src/player.cpp

```diff
@@ -12251,26 +12251,36 @@ int player::encumb( body_part bp ) const
 int player::encumb(body_part bp, double &layers, int &armorenc) const
 {
     int ret = 0;
-    int layer[MAX_CLOTHING_LAYER] = { };
+    // First is the total of encumbrance on the given layer,
+    // and second is the highest encumbrance of any one item on the layer.
+    std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer = {{{0,0},{0,0},{0,0},{0,0},{0,0}}};
```

Comment:
The problem is it's bundled with a class of legitimate warnings, I.e.
you've initialized 3/5 elements in a struct or container to particular
values, most likely because said struct or container grew some elements.
Now the object is partially initialized.
I *think* ```std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer();```
will do it too, it's just the empty brackets that set it off.